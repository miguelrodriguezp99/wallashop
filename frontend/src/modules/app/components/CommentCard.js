import {
	Avatar,
	Card,
	CardBody,
	CardFooter,
	CardHeader,
} from "@nextui-org/react";

import CreateCommentModal from "./modals/CreateCommentModal";

const CommentCard = ({ comment, setCommentCount }) => {
	return (
		<div className="flex justify-center align-center items-center mb-10">
			<Card className="bg-[#171717] w-full max-w-[500px]">
				<CardHeader className="justify-between">
					<div className="flex gap-5">
						<Avatar
							isBordered
							radius="full"
							size="md"
							src={comment.user.avatar}
						/>
						
						<div className="flex flex-col gap-1 items-start justify-center">
							<h4 className="text-small font-semibold leading-none text-default-600">
								{comment.user.userName}
							</h4>
							<h5 className="text-small tracking-tight text-default-400">
								{comment.user.email}
							</h5>
						</div>
					</div>
				</CardHeader>
				<CardBody className="px-3 py-0 text-small text-default-400 text-white">
					<p>{comment.text}</p>
				</CardBody>
				<CardFooter className="gap-3">
					<div className="flex gap-1">
						<CreateCommentModal
							state={comment}
							reply={true}
							setCommentCount={setCommentCount}
						/>
					</div>
				</CardFooter>
			</Card>
		</div>
	);
};

export default CommentCard;
