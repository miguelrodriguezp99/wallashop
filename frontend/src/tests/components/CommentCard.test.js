import React from 'react';
import renderer from 'react-test-renderer';
import CommentCard from '../../modules/app/components/CommentCard';

jest.mock('../../modules/app/components/modals/CreateCommentModal', () => 'CreateCommentModal');
test('renders CommentCard correctly', () => {
  const comment = {
    user: {
      avatar: 'url',
      userName: 'user',
      email: 'a@a.com',
    },
    text: 'comment',
  };

  const setCommentCount = jest.fn();
  const component = renderer.create(<CommentCard comment={comment} setCommentCount={setCommentCount} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});