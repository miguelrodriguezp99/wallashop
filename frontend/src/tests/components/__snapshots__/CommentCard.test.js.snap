// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentCard correctly 1`] = `
<div
  className="flex justify-center align-center items-center mb-10"
>
  <div
    className="flex flex-col relative overflow-hidden height-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none bg-[#171717] w-full max-w-[500px]"
    tabIndex={-1}
  >
    <div
      className="flex p-3 z-10 w-full items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large justify-between"
    >
      <div
        className="flex gap-5"
      >
        <span
          className="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-10 h-10 text-tiny bg-default text-default-foreground rounded-full ring-2 ring-offset-2 ring-offset-background dark:ring-offset-background-dark ring-default"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
        >
          <img
            alt="avatar"
            className="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
            src="url"
          />
        </span>
        <div
          className="flex flex-col gap-1 items-start justify-center"
        >
          <h4
            className="text-small font-semibold leading-none text-default-600"
          >
            user
          </h4>
          <h5
            className="text-small tracking-tight text-default-400"
          >
            a@a.com
          </h5>
        </div>
      </div>
    </div>
    <div
      className="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased px-3 py-0 text-small text-white"
    >
      <p>
        comment
      </p>
    </div>
    <div
      className="p-3 h-auto flex w-full items-center overflow-hidden color-inherit subpixel-antialiased rounded-b-large gap-3"
    >
      <div
        className="flex gap-1"
      >
        <CreateCommentModal
          reply={true}
          setCommentCount={[MockFunction]}
          state={
            Object {
              "text": "comment",
              "user": Object {
                "avatar": "url",
                "email": "a@a.com",
                "userName": "user",
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
